iceoryx_hoofs/source/posix_wrapper/file_lock.cpp

```diff
@@ -60,6 +60,8 @@ cxx::expected<FileLockError> FileLock::initializeFileLock() noexcept
         return cxx::error<FileLockError>(convertErrnoToFileLockError(openCall.get_error().errnum));
     }
 
+    m_isInitialized = true;
```

Comment:
It seems this change will cause the locked file removed unexpected.

Reply:
@qclzdh Good catch! You are right it should be closed only not removed - I fixed that.